sdk/python/kfp/dsl/_for_loop.py

```diff
@@ -0,0 +1,105 @@
+import re
+from typing import List, Union, Dict, Text, Any, Optional
+
+from kfp import dsl
+from kfp.dsl import _metadata
+
+ItemList = List[Union[int, float, str, Dict[Text, Any]]]
+
+
+class LoopArguments(dsl.PipelineParam):
+    """Class representing the arguments that are looped over in a ParallelFor loop in the KFP DSL.
```

Comment:
not a bad idea, but I'm going to skip here since that's a lot of extra documentation to keep in sync, there's nothing complicated to keep track of here, and I rarely if ever see that in Python code in the wild.  Good idea if it violates those conditions, though.